ev3dev2/led.py

```diff
@@ -346,3 +354,214 @@ def all_off(self):
 
         for led in self.leds.values():
             led.brightness = 0
+
+    def reset(self):
+        """
+        Put the LEDs back to their default state of GREEN
+        """
+
+        if not self.leds:
+            return
+
+        self.set_color('LEFT', 'GREEN')
+        self.set_color('RIGHT', 'GREEN')
+
+
+class LedAnimate(threading.Thread):
```

Comment:
That is kind of what we already have it is just via this parent ``LedAnimateThread`` class.  The child classes primarily only have to define their ``run`` method.  I like this model myself...to me it is easier to follow than passing a function pointer to run.

As far as exposing the Thread class, we already do so in ``ev3dev-lang-python-demo``.  Not sure what to do about that for micropython :(

Reply:
Putting aside other pros/cons of the pattern, depending on the API of `Thread` breaks micropython compatibility, so to protect that I would like to avoid tying our public API surface to that of Python threads. The demo repository is its own case as breaking changes don't impact users. I believe it can be done by replacing the superclass with a line in the constructor which instantiates a [Thread](https://docs.python.org/3/library/threading.html#threading.Thread) with `target=self.run` and then adding a class method which starts the thread. It would be nice to also add micropython support with its respective threading API, but we can handle that separately.

Reply:
What if we lazy load the ``thread`` module?  Then we can say that the animation of LEDs is not supported on micropython but it shouldn't break anything else on micropython.

I need to restructure some things to do this...will give it a try and see what it looks like.

Reply:
I'd prefer to design our APIs so that we _can_ support Micropython. As long as the public interface doesn't involve the `Thread` type, that should be fairly easy using the threading primitives available on Micropython.